### ISSUE TITLE
Uploading build for service using provider S3

### ISSUE THREAD

**Issue by boundedinfinity** [rancher/rancher#7746]
Why is rancher attempting to "upload" my private build to S3?  This seems like a bug.

I'm using a `docker-compose.yml` file similar to the following:

```yaml
version: '2'
services:
   my-service:
        image: docker.mycompany.net/mygroup/my-service:1.0.31
        build: .
        ports:
            - 10005:10005
```

I'm using exported environment keys obtained from rancher API section to supply the credentials.  Then when I run:

```
rancher-compose --project-name my-service up --confirm-upgrade -d
```

the service deployment fails with:

```
time="2017-02-08T23:42:04Z" level=info msg="[0/1] [my-service]: Creating "
time="2017-02-08T23:42:04Z" level=info msg="Creating my-service service"
time="2017-02-08T23:42:04Z" level=info msg="Uploading build for my-service using provider S3"
time="2017-02-08T23:42:34Z" level=error msg="Failed Creating my-service : NoCredentialProviders: no valid providers in chain"
time="2017-02-08T23:42:34Z" level=error msg="Failed to start: my-service: NoCredentialProviders: no valid providers in chain"
time="2017-02-08T23:42:34Z" level=fatal msg="NoCredentialProviders: no valid providers in chain"
```

I'm using the `docker-compose.yml` to build the service in my CI and locally, but when I attempt to simple deploy my container it fails.  When I comment out the `build: .` line, it works (as it did using the `docker-compose.yml` v1 syntax).

This seems like a bug for rancher to use a standard docker-compose feature [build](https://docs.docker.com/compose/compose-file/#/build) and [tie it to an Amazon service](https://docs.rancher.com/rancher/v1.4/en/cattle/rancher-compose/build/).

**Rancher Versions:**
Server: v1.3.0
scheduler:rancher/scheduler:v0.5.0

**Docker Version:**
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Wed Jan 18 16:20:26 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Wed Jan 18 16:20:26 2017
 OS/Arch:      linux/amd64
 Experimental: true

**OS and where are the hosts located? (cloud, bare metal, etc):**
cloud / openstack

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node / external db

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
cattle

**Steps to Reproduce:**

**Results:**

**Comment by boundedinfinity** [comment]
NOTE: I've tied this with the following `rancher-compose` versions:

- v0.12.0 (downloaded from the rancher server)
- v0.12.2
- v0.12.2-rc1

**Comment by pestrov** [comment]
Same here. Removing build steps from docker-compose fixes this issue.

**Comment by nfoonf** [comment]
Is it possible in any way to build containers without S3. This seems like a major regression.

**Comment by pestrov** [comment]
Yes, you should build them locally, publish to private registry or dockerhub and then just use that image during the deploy.

**Comment by nfoonf** [comment]
using docker hub is not possible due to possible liability issues. Forcing Users into buying S3 services just to fullfill a task that can be easily made on a running docker-engine seems more than a little bit weird to me, though...

**Comment by pestrov** [comment]
Then you should use a private registry. 
Nobody is forcing to to buy an S3 — you can also just provide a remote link to a source code in you build step. Or just create an image and publish it in any registry.

**Comment by Chris7** [comment]
I have this problem as well. The image has already been built by my CI and uploaded to our private registry. It seems impossible to have a single docker-compose file that provides my build context and also the image to deploy for rancher-compose.

This is my compose:
```
